schemas/json/theme.json

```diff
@@ -1597,6 +1601,8 @@
 						"spacing": {},
 						"typography": {},
 						"filter": {},
+						"shadow": {},
+						"outline": {},
```

Comment:
I wish there was a better way in JSON schema draft-04 to add properties than updating 5 different places in a nearly 2000 line file, but unless we start generating the schema from our own metadata, we'll just have to keep fixing issues like these.

Reply:
Unfortunately the second one doesn't work as [JSON schema validates the sub-schemas in `allOf` independently](https://json-schema.org/understanding-json-schema/reference/object.html?highlight=additionalproperties#extending-closed-schemas). It would validate the types of all of the `stylesProperties` which would pass, and then it would validate `{ "additionalProperties": false }` which says that there should be no properties which would always fail.

It's a limitation of JSON schema in general, and it's one of the reasons some projects, [like the tsconfig JSON schema](https://github.com/microsoft/TypeScript-Website/tree/v2/packages/tsconfig-reference), genreate the schema from metadata instead of writing it by hand.

Reply:
Thank you for the detailed explanation. If there is anything I can do, I will assist you 👍